cmd/commands/server/command_server.go

```diff
@@ -132,10 +133,24 @@ func (cmd *Command) Wait() error {
 // Kill stops server
 func (cmd *Command) Kill() error {
 	cmd.vpnServer.Stop()
-	err := cmd.dialogWaiter.Stop()
+
+	providerId, err := cmd.identityLoader()
+	if err != nil {
+		return err
+	}
+
+	signer := cmd.createSigner(providerId)
+
+	err = cmd.mysteriumClient.UnregisterProposal(providerId.Address, signer)
```

Comment:
You don't need to do anything in this place. Kill command signals openvpn process to exit, openvpn process reports exiting state, and then pinger is stopped and proposal unregistered.

Reply:
Should we depend unregistration after openvpn exit? I think unregistration can be ran separately even before openvpn exit

Reply:
But mostly I miss tests which imitates all those cases we talk here

Reply:
Actually we should call unregister AFTER pinger is stopped. So basically the best place to put Unregister is on command_server.go:117, where stopPinger channel case is handled.

Reply:
So it is:
```
stop service (vpnServer, dialogWaiter, natService) -> stop pinger -> unregister proposal
```